contracts/AppNFTUpgradeable.sol

```diff
@@ -157,7 +165,12 @@ contract AppNFTUpgradeable is Initializable, ERC721Upgradeable, ERC721Enumerable
      * @param uri the uri to set for the token
      * @param appName the name of app to set for the token
      */
-    function safeMintAppNFT(address to, string memory uri, string calldata appName) external whenNotPaused {
+    function safeMintAppNFT(address to, string memory uri, string calldata appName) external payable whenNotPaused {
+
+        if(payForMintFlag){
+            require(msg.value >= mint_fees, "Insufficient fees");
```

Comment:
Use the term `"Insufficient mint fee"` so that people don't confuse it with gas fee.